Bug Report: `sst init` panics with nil pointer dereference on Windows (sst v3.14.0)

## Summary

When running `sst init` in a freshly created Next.js project (created using `bunx create-next-app@latest`), the SST CLI crashes with a panic error related to a `nil pointer dereference`.

## Error Output

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x2781b34]

goroutine 1 [running]:
github.com/sst/sst/v3/pkg/project.Create.func1({0xc000f402c0, 0x16}, {0x0?, 0x0?}, {0xc000f402c0?, 0x16?})
        /home/runner/work/sst/sst/pkg/project/create.go:240 +0x94
io/fs.WalkDir({0x47ff5c0, 0x4883860}, {0xc000f402c0, 0x16}, 0xc0011754e0)
        /opt/hostedtoolcache/go/1.23.8/x64/src/io/fs/walk.go:120 +0xc3
github.com/sst/sst/v3/pkg/project.Create({0x34f2b1b, 0x6}, {0x34ee4df, 0x3})
        /home/runner/work/sst/sst/pkg/project/create.go:239 +0xb06
main.CmdInit(0xc000990400)
        /home/runner/work/sst/sst/cmd/sst/init.go:209 +0x16a5
github.com/sst/sst/v3/cmd/sst/cli.(*Cli).Run(0xc000990400)
        /home/runner/work/sst/sst/cmd/sst/cli/cli.go:90 +0x124
main.run()
        /home/runner/work/sst/sst/cmd/sst/main.go:131 +0x3e5
main.main()
        /home/runner/work/sst/sst/cmd/sst/main.go:55 +0x4ea
```

## Reproduction Steps

1. Create a new Next.js project with Bun:

   ```bash
   bunx create-next-app@latest my-app
   ```

2. Change into the directory:

   ```bash
   cd my-app
   ```

3. Initialize SST:

   ```bash
   sst init
   ```

4. SST crashes with the stack trace shown above.

## Environment

| Name             | Value                                  |
| ---------------- | -------------------------------------- |
| SST Version      | 3.14.0                                 |
| Bun Version      | 1.2.10                                 |
| Node Version     | 22.14.0                                |
| OS               | Windows 10 Pro (Build 19045)           |
| Architecture     | x64-based PC                           |
| CPU              | AMD Ryzen 7 3700X (8-Core, 16 threads) |
| RAM              | 32 GB                                  |
| Shell/Terminal   | Windows Command Prompt                 |
| Project Location | Local drive                            |

--- a/pkg/project/create.go
+++ b/pkg/project/create.go
@@ -235,8 +235,11 @@ func Create(templateName string, home string) ([]string, error) {
 			break
 
 		case "copy":
-			templateFilesPath := filepath.Join("templates", templateName, "files")
+			templateFilesPath := path.Join("templates", templateName, "files")
 			err = fs.WalkDir(platform.Templates, templateFilesPath, func(path string, d fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
 				if d.IsDir() {
 					// Create the directory if it doesn't exist
 					dir := filepath.Join(".", strings.TrimPrefix(path, templateFilesPath))

